server.py

```diff
@@ -911,6 +1024,76 @@ def do_POST(self):
             print(google_email.get_email_history(email_history['historyId']))
             self.send_204()
 
+        elif self.path == "/getObservationStations":
+            try:
+                user_jwt = json_body["jwt"]
+                decoded_jwt = decode_jwt(user_jwt)
+                jwt_user_addr = decoded_jwt["address"]
+            except Exception as e:
+                print(e)
+                pass
+            if isValidEthereumAddress(jwt_user_addr) is False:
+                self.send_400(message='Invalid Ethereum address', explain='Ethereum address pulled from user JWT is not valid')
+                return
+            try:
+                observation_station_numbers = self.db.selectUserStationNumbers_JSON(jwt_user_addr)
+                self.send_200_JSON(json.dumps(observation_station_numbers))
+                return observation_station_numbers
+            except Exception as e:
+                print(e)
+                self.send_500(message='Could not get station information', explain='Query to get observation stations has failed')
+
+        elif self.path == '/generateStation':
+            try:
+                print("TEST")
+                user_jwt = json_body["jwt"]
+                decoded_jwt = decode_jwt(user_jwt)
+                user_addr = decoded_jwt["address"]
+                station_name = json_body["station"]
+                latitude = json_body["latitude"]
+                longitude = json_body["longitude"]
+                altitude = json_body["altitude"]
+                notes = json_body["notes"]
+            except Exception as e:
+                print(e)
+                self.send_400(message='Missing parameter(s)', explain='One or more of the required parameters are missing form the request body')
+                return
+            if isValidEthereumAddress(user_addr) is False:
+                self.send_400(message='message', explain='explanation')
+                return
+            try:
+                user_id = self.db.selectObserverIDFromAddress(user_addr)
+                latest_station = self.db.selectLatestStationID()
+                if latest_station[0:1] == 'T':
+                    station_index = latest_station[1:]
+                    station_index = int(station_index, 36) + 1
+                    # Prevent I or O character
+                    if station_index == 18 or station_index == 24:
+                        station_index = station_index + 1
+                    station_index = numpy.base_repr(station_index, 36)
+                    while len(station_index) < 3:
```

Comment:
Format String T{%03d}

Reply:
@Kmoneal are you going to replace this loop with a format string?

Reply:
This will not work unless the `station_index` is a number but due to using base 36, this is represented as a string to get the alphanumeric version. Instead I am now using right justify on a string with `0` as the padding.